topostats/measure/curvature.py

```diff
@@ -0,0 +1,175 @@
+"""Calculate various curvature metrics for traces."""
+
+from __future__ import annotations
+
+import logging
+
+import numpy as np
+import numpy.typing as npt
+
+from topostats.logs.logs import LOGGER_NAME
+
+LOGGER = logging.getLogger(LOGGER_NAME)
+
+
+def angle_diff_signed(v1: npt.NDArray[np.number], v2: npt.NDArray[np.number]):
+    """
+    Calculate the signed angle difference between two point vecrtors in 2D space.
+
+    Positive angles are clockwise, negative angles are counterclockwise.
+
+    Parameters
+    ----------
+    v1 : npt.NDArray[np.number]
+        First vector.
+    v2 : npt.NDArray[np.number]
+        Second vector.
+
+    Returns
+    -------
+    float
+        The signed angle difference in radians.
+    """
+    if v1.shape != (2,) or v2.shape != (2,):
+        raise ValueError("Vectors must be of shape (2,)")
+
+    angle = np.arctan2(v2[1], v2[0]) - np.arctan2(v1[1], v1[0])
+    if angle > np.pi:
+        angle -= 2 * np.pi
+    elif angle < -np.pi:
```

Comment:
What if `angle == np.pi` (rare but you never know!)? I can see `angle` is then returned without subtracting/adding `2 * np.pi` but is that the desired behaviour?

Reply:
Very thorough reading there! But I believe this is intended. The domain for the angles is designed to be `[-np.pi, np.pi]` inclusive.